### PR TITLE
Added s3_quota_notice.py: py REST version of s3_quota_notice.sh script.

### PR DESCRIPTION
**Pull request by k0ste**
Hello.
I'm introducing python version inspired by `s3-quota-notice` shell script. The main idea is use this script in any environment with ability to connect to rgw's from any place, and also possibility to use in any company (I mean the e-mail text message) without necessity to setup vlans, keyrings, `ceph.conf`, `.mailrc`, etc

* Use RadosGW Admin API instead radosgw-admin binary;
* Use SMTP connection instead sendmail binary;
* Support SMTP AUTH, STARTTLS;
* Added simple logging to determine what user is reached his quota;
* Customizable e-mail settings: defaults is still for Cern;

Requirements ([list user admin OP API feature](https://tracker.ceph.com/issues/37073)):

* [Luminous 12.2.13](https://tracker.ceph.com/issues/42895)+;
* [Mimic 13.2.9](https://tracker.ceph.com/issues/42896)+;
* [requests-aws](//github.com/tax/python-requests-aws);

Tested on:

* python 3.6 on CentOS 7;
* python 3.8;

### CODE REVIEW

File: tools/s3_quota_notice.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+        if user_quota:
+            return user_quota
+
+        return
+
+    def check_rgw_user_quota(self, quota):
+        """
+        Method checks if qouta enabled, than checks that quota size is defined,
+        e.g. qouta is not unlimited.
```
Comment (thmour) [inline tools/s3_quota_notice.py]: qouta --> quota

File: tools/s3_quota_notice.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+        if self.smtp_login:
+            try:
+                self.smtp.login(self.smtp_login, self.smtp_password)
+            except Exception as e:
+                logging.error("SMTP Error: {0}".format(e))
+                exit(2)
+
+
+    def qouta_reached_message(self, uid, quota, stats, email):
```
Comment (thmour) [inline tools/s3_quota_notice.py]: qouta --> quota

File: tools/s3_quota_notice.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        if rgw_users:
+            for uid in rgw_users:
+                quota = self.get_rgw_user_quota(uid)
+                email, stats = self.get_rgw_user_info(uid)
+
+                if self.check_rgw_user_quota(quota) and \
+                    self.check_quota_usage(stats['size_actual'], quota['max_size']):
+                        logging.info("Quota for uid '{0}' is reached. Current usage is {1}%.".format(uid, \
+                            self.quota_usage_percent(stats['size_actual'], quota['max_size'])))
+                        self.qouta_reached_message(uid, quota, stats, email)
```
Comment (thmour) [inline tools/s3_quota_notice.py]: qouta --> quota

File: tools/s3_quota_notice.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+    def get_rgw_users(self):
+        """
+        API request to get users.
+        """
+
+        rgw_users = self.make_rgw_query("user", "?list")
+
+        if rgw_users:
+            return rgw_users['keys']
```
Comment (thmour) [inline tools/s3_quota_notice.py]: ceph has no `s3url/admin/user?list` uri, the correct value would be `"metadata/user"` on the first parameter instead.
what you get back is an array of the user uids, e.g. `['user1','user2']` and not a dict.

Comment (k0ste) [reply tools/s3_quota_notice.py]: `user?list` is a new for admin API. I've posted URL's to tracker in description.

**Comment by k0ste** [comment]
> check for typos
Thanks, fixed.

File: tools/s3_quota_notice.py (hunk 4 above)
Comment (thmour) [reply tools/s3_quota_notice.py]: if `metadata/user` works for luminous mimic and nautilus, i would suggest we use that instead. If not, try to implement this functionality of listing users for all mentioned versions, or exit the script if the ceph version is not the required one. It's not good to run into weird responses while running the script.

Comment (k0ste) [reply tools/s3_quota_notice.py]: Indeed. Can you provide example of `metadata/user` query? I don't see any documentation for this method. Account need some read caps for this?
Thanks.

Comment (thmour) [reply tools/s3_quota_notice.py]: the account needs 'metadata=read or *' caps, my first comment shows you exactly what rgw_users is after changing `user` to `metadata/user`

Comment (k0ste) [reply tools/s3_quota_notice.py]: Added fallback query.

File: tools/s3_quota_notice.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+
+    def get_rgw_users(self):
+        """
+        API request to get users.
+        """
+
+        rgw_users = self.make_rgw_query("user", "?list")
+
+        if "keys" in rgw_users:
```
Comment (thmour) [inline tools/s3_quota_notice.py]: rgw_users returns `None` if it fails, add `rgw_users and` after the if

File: tools/s3_quota_notice.py [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+            return rgw_users['keys']
+        else:
+            # Compat with old Ceph versions (pre 12.2.3/13.2.9)
+            rgw_metadata_users = self.make_rgw_query("metadata", "/user")
```
Comment (thmour) [inline tools/s3_quota_notice.py]: your second parameter is the parameters of the query, technically the `/user` is still before the `?` point of the uri, the endpoint is `metadata/user, either keep the second parameter as an empty string or add the empty string as a default parameter

**Pull request merged**